website/docs/core-concepts/data-access-and-binding/displaying-data-read/README.md

```diff
@@ -1,34 +1,77 @@
+
+```javascript
+{{Input1.text}}
+```
+
+Enter a value in `Input1` and see how the value updates in `Input2`.
+
+**Example 3:** suppose you have two input widgets and one button widget named `Input1`, `Input2`, and `Button1`, respectively. This example shows how to update `Input2` with the value in `Input1` on the button click. Here, the [storeValue()](/reference/appsmith-framework/widget-actions/store-value) function is used.
+
+Paste the following code in the `onClick` event of `Button1`. 
+
+```javascript
+{{storeValue('inputData',Input1.text)}}
+```
+Paste the following code in the `Default Value` property of Input2.
+```javascript
+{{appsmith.store.inputData}}
+```
+
+Enter a value in `Input1`. On button click, the value updates in `Input2`.
+
+### Display data from async JS function
 
-## Transforming Data
+Widgets have fields/properties where you can bind data or trigger actions.
 
-You can use Javascript inside to transform Query data when binding it to a property. Let us take an example of a Query that returns an array of objects that need to be populated in a [dropdown](/reference/widgets/select.md). Directly binding the data will lead to an error as shown below
+**Sync fields** are properties that expect input or data. For example, for an Input widget, properties such as `Default Value`, `Max Characters`, `Regex`, and `Error Message` expect input and are sync fields.
 
-A [select](/reference/widgets/select.md) needs an Array\<label, value> in its option field, so to connect this data to a dropdown, we need to transform the data in the [dropdown options property.](/reference/widgets/select#widget-properties)
+**Async fields** are properties that can trigger an action or perform an operation. For example, the properties like `OnTextChanged` and `OnSubmit` of an input widget are async fields. You can use these properties to execute an [action](/reference/appsmith-framework/widget-actions), [Query](/core-concepts/data-access-and-binding/querying-a-database#running-a-query) or a function within a [JS object](/core-concepts/writing-code/javascript-editor-beta).
+
+To display the response from an asynchronous JS function in a synchronous field, you need to retrieve it using the  `.data` property as shown below
+
+```javascript
+{{JSObjectName.functionName.data}}
+```
+
+ <VideoEmbed host="youtube" videoId="yn_8gs5w04g" title="Display response from async function in widget field" caption="Display response from async function in widget field"/> 
+
+### Display data from sync JS function
+
+To display the response from a synchronous JS function in a widget field, call the function inside the JS Object as shown below:
+
+``` javascript
+{{JSObjectName.functionName()}}
+```
+
+## Transforming data
+
+You can use Javascript to transform Query data when binding it to a property. For example, consider a Query that returns an array of objects as shown below:  
```

Comment:
```suggestion
Each widget property has a specific data type that it validates its value against. If the data type mismatches, it will throw an error. You can use JavaScript to transform the data when binding it to a property. For example, consider a query that returns an array of objects, as shown below:  
```